tests/suites/os/tests/config-json/index.js

```diff
@@ -306,44 +306,53 @@ module.exports = {
 		{
 			title: 'udevRules test',
 			run: async function(test) {
+				const context = this.context.get();
 				const rule = {
 					99: 'ENV{ID_FS_LABEL_ENC}=="resin-boot", SYMLINK+="disk/test"',
 				};
 
-				await this.context
-					.get()
-					.worker.executeCommandInHostOS(
-						`tmp=$(mktemp)&&cat /mnt/boot/config.json | jq '.os.udevRules=${JSON.stringify(
-							rule,
-						)}' > $tmp&&mv "$tmp" /mnt/boot/config.json`,
-						this.context.get().link,
+				test.comment('Adding udev rule to config.json');
+				return context.worker.executeCommandInHostOS(
+					[
+						`tmp=$(mktemp)`,
+						`&&`, `jq`, `'.os.udevRules=${JSON.stringify(rule)}'`, `/mnt/boot/config.json`,
+						`>`, `$tmp`, `&&`, `mv`, `$tmp`, `/mnt/boot/config.json`,
+					].join(' '),
+					context.link,
+				).then(() => {
+					test.comment('Restarting os-udevrules.service');
+					return context.worker.executeCommandInHostOS(
+						'systemctl restart os-udevrules.service',
```

Comment:
Should we actually be making this service part of the `config-json.target` group to be restarted automatically? We have several services following this pattern already.

Reply:
That's good feedback, I wasn't aware of this. I think it would be better scoped in another PR, however.

Reply:
Maybe something to discuss as a separate PR...